### ISSUE TITLE
Weird event subscription logic?

### ISSUE THREAD

**Issue by dvv** [filp/homeassistant-ws#6]
Running

```js
const onStateChanged = async (ev) => {
  console.debug("*EVT", ev)
})
async function main () {
  client = await hass({
    token: "bebebe",
  })
  await client.onStateChanged(onStateChanged)
  await client.onAnyEvent(ev => {
    // notice nothing's here
  })
}

main()
```

gives me **two identical** `"*EVT ..."` lines for each state change.

How to cope with?

TIA

**Comment by filp** [comment]
Hey @dvv - thanks for the issue report, and apologies in advance for the late response. I'll take a look at this soon to try and replicate it, but it doesn't strike me as unlikely that I didn't properly account for a case where you'd use both `onStateChanged` and `onAnyEvent` at the same time.

Hope the upcoming fix will still be useful for your use case, and thanks again for reporting it!

**Comment by filp** [comment]
Resolved in a56e16c with the simplified event handling logic.
